### PR TITLE
[OYPD-XXX] Embedded GroupEx Pro Schedule

### PR DESCRIPTION
**Pull request by duozersk**
Jira issue: https://propeople-us.atlassian.net/browse/YLI-141

See the TODO comment in the code changes - the block settings form is not saving the values.

Steps to review:
- [ ] login as admin
- [ ] go to edit some landing page, like ```/join```
- [ ] verify you can add "Embedded GroupEx Pro Schedule" paragraph to the "CONTENT AREA" - add it there
- [ ] verify you don't have this paragraph in "HEADER AREA" and "BOTTOM AREA"
- [ ] save the landing page
- [ ] view it and verify you see the added paragraph like on the screen shot below:
screen shot 2017-04-11 at 18 06 19

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build915 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build915 |
| Installation process | http://install.openy-dev.ffwua.com/build915/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/915/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build915/profiles/contrib/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build916 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build916 |
| Installation process | http://install.openy-dev.ffwua.com/build916/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/916/

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build917 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build917 |
| Installation process | http://install.openy-dev.ffwua.com/build917/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/917/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build916/profiles/contrib/openy/build/reports/behat

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build917/profiles/contrib/openy/build/reports/behat

File: modules/custom/embedded_groupexpro_schedule/src/Plugin/Block/EmbeddedGroupExProScheduleBlock.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $config = $this->getConfiguration();
+
+    // TODO: remove this fallback; block configuration is not saved when
+    // configuring it as a plugin reference field on paragraph entity type.
+    if (empty($config['account'])) {
+      $config['account'] = 611;
```
Comment (ddrozdik) [inline modules/custom/embedded_groupexpro_schedule/src/Plugin/Block/EmbeddedGroupExProScheduleBlock.php:52]: We should not have account here or should use not related to YMCA, maybe demo content. Let's review account 1

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build919 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build919 |
| Installation process | http://install.openy-dev.ffwua.com/build919/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/919/

**Review by ddrozdik — CHANGES_REQUESTED** [review]

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build919/profiles/contrib/openy/build/reports/behat

File: modules/custom/embedded_groupexpro_schedule/src/Plugin/Block/EmbeddedGroupExProScheduleBlock.php (hunk 1 above)
Comment (ddrozdik) [reply modules/custom/embedded_groupexpro_schedule/src/Plugin/Block/EmbeddedGroupExProScheduleBlock.php:52]: Or we should provide a field, which will allow overriding account

Comment (duozersk) [reply modules/custom/embedded_groupexpro_schedule/src/Plugin/Block/EmbeddedGroupExProScheduleBlock.php:52]: We do have the block configuration form with account textfield - it just doesn't save the values.

**Comment by alexschedrov** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/1104/

**Build Log**
*last 1000 lines*

 ```
GitHub pull request #290 of commit 866e341cda78ea3238dc7b6fdca579feb33c3662.
GitHub pull request #290 of commit 866e341cda78ea3238dc7b6fdca579feb33c3662.
Building in workspace /var/lib/jenkins/jobs/PR_BUILDER/workspace
Cloning the remote Git repository
Using shallow clone
Cloning repository https://github.com/ymcatwincities/openy.git
 > git init /var/lib/jenkins/jobs/PR_BUILDER/workspace/sourcecode # timeout=10
Fetching upstream changes from https://github.com/ymcatwincities/openy.git
 > git --version # timeout=10
using .gitcredentials to set credentials
 > git config --local credential.helper store --file=/tmp/git5434677729207330292.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/ymcatwincities/openy.git +refs/heads/*:refs/remotes/origin/* --depth=1 # timeout=5
 > git config --local --remove-section credential # timeout=10
 > git config remote.origin.url https://github.com/ymcatwincities/openy.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/ymcatwincities/openy.git # timeout=10
Cleaning workspace
 > git rev-parse --verify HEAD # timeout=10
No valid HEAD. Skipping the resetting
 > git clean -fdx # timeout=10
Fetching upstream changes from https://github.com/ymcatwincities/openy.git
using .gitcredentials to set credentials
 > git config --local credential.helper store --file=/tmp/git5917696106948764398.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/ymcatwincities/openy.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=5
 > git config --local --remove-section credential # timeout=10
 > git rev-parse 866e341cda78ea3238dc7b6fdca579feb33c3662^{commit} # timeout=10
 > git branch -a --contains 866e341cda78ea3238dc7b6fdca579feb33c3662 # timeout=10
 > git rev-parse remotes/origin/pr/290/head^{commit} # timeout=10
 > git rev-parse remotes/origin/pr/290/merge^{commit} # timeout=10
Checking out Revision 866e341cda78ea3238dc7b6fdca579feb33c3662 (origin/pr/290/head, origin/pr/290/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 866e341cda78ea3238dc7b6fdca579feb33c3662
First time build. Skipping changelog.
[workspace] $ /bin/sh -xe /tmp/hudson4755570605843439946.sh
+ cd /var/lib/jenkins/jobs/PR_BUILDER/workspace/vm
+ git pull
Already up-to-date.
[workspace] $ ansible-playbook /var/lib/jenkins/jobs/PR_BUILDER/workspace/vm/cibox/jobs/deploy_code.yml -i /tmp/inventory2339208789665781313.ini -f 5 -e workspace=/var/lib/jenkins/jobs/PR_BUILDER/workspace -e build_number=1104 -e server_docroot_folder=/var/www -e deploy_type=local -e profile_source=/var/lib/jenkins/jobs/PR_BUILDER/workspace/sourcecode -e source=/var/lib/jenkins/jobs/PR_BUILDER/workspace/vm/docroot -e profile_destination=profiles/contrib/openy -e build_folder_prefix=build -i localhost, --connection=local

PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [../custom/cibox-code-deploy | Generate commentinfo.md file] ************ 
changed: [localhost]

TASK: [../custom/cibox-code-deploy | Make sure folder is not exist] *********** 
ok: [localhost]

TASK: [../custom/cibox-code-deploy | Sync files into docroot] ***************** 
changed: [localhost]

TASK: [../custom/cibox-code-deploy | Sync OpenY profile] ********************** 
changed: [localhost]

TASK: [../custom/cibox-code-deploy | Folder permissions] ********************** 
changed: [localhost]

PLAY RECAP ******************************************************************** 
localhost                  : ok=6    changed=4    unreachable=0    failed=0   

[workspace] $ ansible-playbook /var/lib/jenkins/jobs/PR_BUILDER/workspace/vm/cibox/jobs/build.yml -i /tmp/inventory7433660780944904354.ini -f 5 -e workspace=/var/lib/jenkins/jobs/PR_BUILDER/workspace -e build_number=1104 -e server_docroot_folder=/var/www -e build_folder_prefix=build -i localhost, --connection=local

PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [Folder permissions] **************************************************** 
changed: [localhost]

PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

[workspace] $ ansible-playbook /var/www/build1104/reinstall.yml -f 5 -e php_env_vars=APP_ENV=dev -e mysql_user=root -e mysql_password=root -e mysql_db=drupal1104 -e drupal_folder=/var/www/build1104 -e site_url=http://openy-dev.ffwua.com/build1104 -e pp_environment=default -e run_reinstall=true -i localhost, --connection=local

PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [Include specific variables per environment] **************************** 
ok: [localhost]

TASK: [Getting timestamp for backup] ****************************************** 
skipping: [localhost]

TASK: [Getting name for backup] *********************************************** 
skipping: [localhost]

TASK: [Creating backup folder for environment] ******************************** 
skipping: [localhost]

TASK: [Making regular backup from {{ local_backup_environment }} environment] *** 
skipping: [localhost]

TASK: [Make new backup as latest] ********************************************* 
skipping: [localhost]

TASK: [Set writable cache dir] ************************************************ 
ok: [localhost]

TASK: [Set writable tmp dir] ************************************************** 
changed: [localhost]

TASK: [Set writable backup dir] *********************************************** 
ok: [localhost]

TASK: [Set writable default dir] ********************************************** 
changed: [localhost]

TASK: [Set writable files dir] ************************************************ 
changed: [localhost]

TASK: [Set writable private dir] ********************************************** 
changed: [localhost]

TASK: [Start the MySQL service] *********************************************** 
ok: [localhost]

TASK: [Getting name for settings (WINDOWS)] *********************************** 
skipping: [localhost]

TASK: [Getting name for settings (UNIX)] ************************************** 
skipping: [localhost]

TASK: [Remove settings.php] *************************************************** 
skipping: [localhost]

TASK: [Creating settings.php] ************************************************* 
skipping: [localhost]

TASK: [Adding database variable to settings.php] ****************************** 
skipping: [localhost]

TASK: [Add general pre-settings to settings.php] ****************************** 
skipping: [localhost]

TASK: [Add environment specific pre-settings] ********************************* 
skipping: [localhost] => (item={'status': True, 'name': '$settings["hash_salt"]', 'value': 'g1W4KZgWCeWh7Af9_gSb9D9HOiNaBdWPvQD-8x0yWpcgNpTBppaHR5VNGQBGvh2k8P8uWa1V8w'})
skipping: [localhost] => (item={'status': False, 'name': '$config_directories["staging"]', 'value': 'sites/default/config/staging'})
skipping: [localhost] => (item={'status': True, 'name': '$config_directories["sync"]', 'value': 'sites/default/config/sync'})
skipping: [localhost] => (item={'status': True, 'name': '$settings["install_profile"]', 'value': u'openy'})
skipping: [localhost] => (item={'status': True, 'name': '$settings["file_public_path"]', 'value': u'sites/default/files'})
skipping: [localhost] => (item={'status': True, 'name': '$config["system.file"]["path.temporary"]', 'value': '/tmp'})

TASK: [Copy settings back (WINDOWS)] ****************************************** 
skipping: [localhost]

TASK: [Droping drupal database] *********************************************** 
skipping: [localhost]

TASK: [Creating drupal database] ********************************************** 
skipping: [localhost]

TASK: [Downloading latest backup to local environment] ************************ 
skipping: [localhost]

TASK: [Removing possible abandoned dump] ************************************** 
skipping: [localhost]

TASK: [Importing from remote backup] ****************************************** 
skipping: [localhost]

TASK: [Copy latest backup to build folder] ************************************ 
skipping: [localhost]

TASK: [Importing from local backup] ******************************************* 
skipping: [localhost]

TASK: [Remove local backup from build folder] ********************************* 
skipping: [localhost]

TASK: [Build project from drupal-org.make] ************************************ 
skipping: [localhost]

TASK: [Clear drush cache] ***************************************************** 
skipping: [localhost]

TASK: [Rebuilding drupal registry] ******************************************** 
skipping: [localhost]

TASK: [Clear Drupal cache] **************************************************** 
skipping: [localhost]

TASK: [Updating database] ***************************************************** 
skipping: [localhost]

TASK: [Apply pending entity schema updates] *********************************** 
skipping: [localhost]

TASK: [Download stage file proxy] ********************************************* 
skipping: [localhost]

TASK: [Enable stage file proxy] *********************************************** 
skipping: [localhost]

TASK: [Set configuration for stage file proxy] ******************************** 
skipping: [localhost]

TASK: [Set folder for stage file proxy] *************************************** 
skipping: [localhost]

TASK: [Updates listing] ******************************************************* 
skipping: [localhost]

TASK: [Droping drupal database] *********************************************** 
ok: [localhost]

TASK: [Creating drupal database] ********************************************** 
changed: [localhost]

TASK: [Remove settings.php] *************************************************** 
ok: [localhost]

TASK: [Build project from drupal-org.make] ************************************ 
changed: [localhost]

TASK: [Installing drupal] ***************************************************** 
failed: [localhost] => {"changed": true, "cmd": "APP_ENV=dev drush -y si openy --db-url=mysql://root:root@127.0.0.1:/drupal1104 --account-name=admin --account-pass=openy --uri=http://openy-dev.ffwua.com/build1104", "delta": "0:00:20.468559", "end": "2017-05-05 13:12:36.855773", "rc": 1, "start": "2017-05-05 13:12:16.387214", "warnings": []}
stderr: Starting Drupal installation. This takes a while. Consider using the [ok]
--notify global option.
ArgumentCountError: Too few arguments to function Drupal\Core\Config\ConfigInstaller::__construct(), 5 passed in /var/www/build1104/profiles/contrib/openy/modules/contrib/features/src/FeaturesConfigInstaller.php on line 56 and exactly 6 expected in /var/www/build1104/core/lib/Drupal/Core/Config/ConfigInstaller.php on line 85 #0 /var/www/build1104/profiles/contrib/openy/modules/contrib/features/src/FeaturesConfigInstaller.php(56): Drupal\Core\Config\ConfigInstaller->__construct(Object(Drupal\Core\Config\ConfigFactory), Object(Drupal\Core\Config\CachedStorage), Object(Drupal\Core\Config\TypedConfigManager), Object(Drupal\Core\Config\ConfigManager), Object(Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher))
#1 [internal function]: Drupal\features\FeaturesConfigInstaller->__construct(Object(Drupal\Core\ProxyClass\Config\ConfigInstaller), Object(Drupal\features\FeaturesManager), Object(Drupal\Core\Config\ConfigFactory), Object(Drupal\Core\Config\CachedStorage), Object(Drupal\Core\Config\TypedConfigManager), Object(Drupal\Core\Config\ConfigManager), Object(Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher))
#2 /var/www/build1104/vendor/symfony/dependency-injection/ContainerBuilder.php(928): ReflectionClass->newInstanceArgs(Array)
#3 /var/www/build1104/vendor/symfony/dependency-injection/ContainerBuilder.php(468): Symfony\Component\DependencyInjection\ContainerBuilder->createService(Object(Symfony\Component\DependencyInjection\Definition), 'config.installe...')
#4 /var/www/build1104/core/lib/Drupal.php(158): Symfony\Component\DependencyInjection\ContainerBuilder->get('config.installe...')
#5 /var/www/build1104/core/lib/Drupal/Core/Extension/ModuleInstaller.php(242): Drupal::service('config.installe...')
#6 /var/www/build1104/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83): Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#7 /var/www/build1104/core/includes/install.core.inc(1800): Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array, false)
#8 /var/www/build1104/core/includes/batch.inc(252): _install_module_batch('features', 'Features', Array)
#9 /var/www/build1104/core/includes/form.inc(874): _batch_process()
#10 /var/www/build1104/core/includes/install.core.inc(619): batch_process(Object(Drupal\Core\Url), Object(Drupal\Core\Url))
#11 /var/www/build1104/core/includes/install.core.inc(540): install_run_task(Array, Array)
#12 /var/www/build1104/core/includes/install.core.inc(117): install_run_tasks(Array)
#13 /usr/local/src/drush/includes/drush.inc(726): install_drupal(Object(Composer\Autoload\ClassLoader), Array)
#14 /usr/local/src/drush/includes/drush.inc(711): drush_call_user_func_array('install_drupal', Array)
#15 /usr/local/src/drush/commands/core/drupal/site_install.inc(80): drush_op('install_drupal', Object(Composer\Autoload\ClassLoader), Array)
#16 /usr/local/src/drush/commands/core/site_install.drush.inc(249): drush_core_site_install_version('openy', Array)
#17 /usr/local/src/drush/includes/command.inc(422): drush_core_site_install('openy')
#18 /usr/local/src/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#19 /usr/local/src/drush/includes/command.inc(199): drush_command('openy')
#20 /usr/local/src/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#21 /usr/local/src/drush/includes/preflight.inc(66): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#22 /usr/local/src/drush/drush.php(12): drush_main()
#23 {main}
Drush command terminated abnormally due to an unrecoverable error.   [error]
stdout: You are about to create a /var/www/build1104/sites/default/settings.php file and DROP all tables in your 'drupal1104' database. Do you want to continue? (y/n): y
ArgumentCountError: Too few arguments to function Drupal\Core\Config\ConfigInstaller::__construct(), 5 passed in /var/www/build1104/profiles/contrib/openy/modules/contrib/features/src/FeaturesConfigInstaller.php on line 56 and exactly 6 expected in Drupal\Core\Config\ConfigInstaller->__construct() (line 85 of /var/www/build1104/core/lib/Drupal/Core/Config/ConfigInstaller.php).

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=14   changed=6    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by alexschedrov** [comment]
Work is continued in #290.

**Pull request closed without merging**